client/src/scene/SceneGameArena.ts

```diff
@@ -162,13 +162,33 @@ export class SceneGameArena extends Phaser.Scene {
     private updateUserInput(scene: SceneGameArena) {
         let moved = false;
         if (scene.keys.a.isDown || scene.keys.left.isDown) {
-            moved = scene.gameState.currentTetromino.move(-1);
+            moved = scene.gameState.currentTetromino.moveIfCan(
+                scene.gameState.board,
+                (tetro) => {
+                    tetro.position[1] -= 1;
+                }
+            );
         } else if (scene.keys.d.isDown || scene.keys.right.isDown) {
-            moved = scene.gameState.currentTetromino.move(1);
+            moved = scene.gameState.currentTetromino.moveIfCan(
+                scene.gameState.board,
+                (tetro) => {
+                    tetro.position[1] += 1;
+                }
+            );
         } else if (scene.keys.q.isDown || scene.keys.z.isDown) {
-            moved = scene.gameState.currentTetromino.rotateCCW();
+            moved = scene.gameState.currentTetromino.moveIfCan(
+                scene.gameState.board,
+                (tetro) => {
+                    tetro.rotateCCW();
+                }
```

Comment:
as `rotateCCW` and `rotateCW` are helper functions anyways, why not make them static so they can be passed in directly here (without needing the lambda function).
Would be cleaner, and perhaps a pattern we'd also want to follow for movement

Reply:
Also replaced falling and moving left/right with `fall` and [`slide`](https://github.com/ECE493G14Capstone/Capstone/pull/63/files#diff-f4fa39ef49d7942d6ba503bfd24cb4532d7efda51e965febfec0d8c53a9e48faR198)